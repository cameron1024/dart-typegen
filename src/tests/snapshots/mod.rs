// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
