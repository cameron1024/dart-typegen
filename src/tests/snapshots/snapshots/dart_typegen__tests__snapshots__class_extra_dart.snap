---
source: src/tests/snapshots/mod.rs
expression: dart
---
final class Foo {
  final String x;

  const Foo({required this.x});

  static FooBuilder builder({required String x}) => FooBuilder(x: x);
  FooBuilder toBuilder() => FooBuilder(x: x);

  Map<String, dynamic> toJson() => {"x": x};
  factory Foo.fromJson(Map<String, dynamic> json) =>
      Foo(x: json["x"] as String);

  @override
  String toString() =>
      "Foo("
      "x: $x"
      ")";
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) {
      return true;
    }
    if (other is! Foo) {
      return false;
    }
    if (x != other.x) return false;
    return true;
  }

  @override
  int get hashCode => Object.hashAll([x]);

  void baz() {
    print(x);
  }
}

/// Builder class for [Foo]
final class FooBuilder {
  String x;

  FooBuilder({required this.x});

  Foo build() => Foo(x: x);
}
